spec/app_support/order_row_importer_spec.rb

```diff
@@ -0,0 +1,271 @@
+  let(:order_date) { "column5" }
+  let(:fulfillment_date) { "column6" }
+  let(:errors) { "column7" }
+  let(:notes) { "column8" }
+
+  describe "#import" do
+    shared_examples_for "an order was created" do
+      it "creates an order" do
+        expect { subject.import }.to change(Order, :count).by(1)
+      end
+    end
+
+    shared_examples_for "an order was not created" do
+      it "does not create an order" do
+        expect { subject.import }.not_to change(Order, :count)
+      end
+
+      it "does not add an order_detail" do
+        expect { subject.import }.not_to change(OrderDetail, :count)
+      end
+    end
+
+    context "when the fulfillment date is invalid" do
+      it_behaves_like "an order was not created"
+
+      it "flags the fulfillment date as invalid" do
+        subject.import
+        expect(subject.errors)
+          .to include_a_string_matching(/Invalid Fulfillment Date/)
+      end
+    end
+
+    context "when the order date is invalid" do
+      it_behaves_like "an order was not created"
+
+      it "flags the order date as invalid" do
+        subject.import
+        expect(subject.errors)
+          .to include_a_string_matching(/Invalid Order Date/)
+      end
+    end
+
+    context "when the user is invalid" do
+      context "when looking up by username" do
+        it_behaves_like "an order was not created"
+
+        it "flags the username as invalid" do
+          subject.import
+          expect(subject.errors)
+            .to include_a_string_matching(/Invalid username/)
+        end
+      end
+
+      context "when looking up by email address" do
+        it_behaves_like "an order was not created"
+
+        it "flags the username as invalid" do
+          subject.import
+          expect(subject.errors)
+            .to include_a_string_matching(/Invalid username/)
+        end
+      end
+    end
+
+    context "when the product name is invalid" do
+      it_behaves_like "an order was not created"
+
+      it "flags the product name as invalid" do
+        subject.import
+        expect(subject.errors)
+          .to include_a_string_matching(/Couldn't find product by name/)
+      end
+    end
+
+    context "when the product is a service" do
+      let(:chart_string) { account.account_number }
+      let(:price_group) { create(:price_group, facility: facility) }
+      let(:product) { service }
+      let(:product_name) { product.name }
+      let(:username) { user.username }
+
+      before(:each) do
+        create(:user_price_group_member, user: user, price_group: price_group)
+        product.service_price_policies.create(
+          attributes_for(:service_price_policy, price_group: price_group)
+        )
+      end
+
+      context "and it requires a survey" do
+        before { Service.any_instance.stub(:active_survey?).and_return(true) }
+
+        it_behaves_like "an order was not created"
+
+        it "flags the service as requiring a survey" do
```

Comment:
We could do a shared example with a parameter:
`it_behaves_like "has error message", /Service requires survey/`.
That could shorten the spec and make it easier to read I think. Just a thought.